tensordict/utils.py

```diff
@@ -2260,3 +2268,45 @@ def __missing__(self, key):
         value = self.fun(key)
         self[key] = value
         return value
+
+
+def _unravel_key_to_tuple(key):
+    if isinstance(key, str):
+        return (key,)
+    if not isinstance(key, tuple):
+        return ()
+    return tuple(subk for k in key for subk in _unravel_key_to_tuple(k))
+
+
+def _slice_indices(index: slice, len: int):
```

Comment:
this can go into PyTorch core `_dynamo/polyfill.py` and then inline it in dynamo.